warehouse/query-core/src/main/java/datawave/query/planner/QueryOptionsSwitch.java

```diff
@@ -0,0 +1,41 @@
+package datawave.query.planner;
+
+import com.google.common.collect.Sets;
+import datawave.query.Constants;
+import datawave.query.config.ShardQueryConfiguration;
+import datawave.query.iterator.QueryOptions;
+import datawave.util.StringUtils;
+
+import java.util.Map;
+
+public class QueryOptionsSwitch {
+    
+    public static void apply(Map<String,String> optionsMap, ShardQueryConfiguration config) {
+        for (Map.Entry<String,String> entry : optionsMap.entrySet()) {
+            String key = entry.getKey();
+            String value = entry.getValue();
+            switch (key) {
+                case QueryOptions.INCLUDE_GROUPING_CONTEXT:
+                    config.setIncludeGroupingContext(Boolean.parseBoolean(value));
+                    break;
+                case QueryOptions.HIT_LIST:
+                    config.setHitList(Boolean.parseBoolean(value));
+                    break;
+                case QueryOptions.LIMIT_FIELDS:
+                    String[] lf = StringUtils.split(value, Constants.PARAM_VALUE_SEP);
+                    config.setLimitFields(Sets.newHashSet(lf));
+                    break;
+                case QueryOptions.TYPE_METADATA_IN_HDFS:
```

Comment:
users should not be able to override TYPE_METADATA_IN_HDFS or DISABLE_EVALUATION.  Perhaps if the user is an admin.....

Reply:
I agree about DISABLE_EVALUATION because it could cause incorrect results.
I believe there is a requirement (perhaps not a permanent requirement) to allow users to select the type.metadata.in.hdfs optimization be turned on or off. Query results are expected to be the same either way.

Reply:
ok